packages/dds/tree/src/test/domains/json/jsonCursor.spec.ts

```diff
@@ -64,35 +61,39 @@ describe("JsonCursor", () => {
     });
 
     describe("keys", () => {
+        const getFieldKey = (cursor: ITreeCursorNew) => cursor.getFieldKey();
+        const getKeysAsSet = (cursor: ITreeCursorNew) =>
+            new Set(mapCursorFields(cursor, getFieldKey));
+
         it("object", () => {
-            assert.deepEqual([...new JsonCursor({}).keys], []);
-            assert.deepEqual([...new JsonCursor({ x: {} }).keys], ["x"]);
+            assert.deepEqual(getKeysAsSet(singleJsonCursor({})), new Set());
+            assert.deepEqual(getKeysAsSet(singleJsonCursor({ x: {} })), new Set(["x"]));
             assert.deepEqual(
-                new Set(new JsonCursor({ x: {}, test: 6 }).keys),
+                getKeysAsSet(singleJsonCursor({ x: {}, test: 6 })),
                 new Set(["x", "test"]),
             );
         });
 
         it("array", () => {
             // TODO: should empty arrays report this key?
```

Comment:
The cursor API currently doesn't clarify this. We should fix that. I think explicitly documenting that only non-empty fields will be included in the iteration is reasonable. The other implementations attempt to ensure that, and your own cursorToJsonObject asserts if it finds an empty field on an object, so we should probably require that in the cursor API, fix empty arrays here, and update the tests accordingly.

If you want to update the existing generic cursor tests and API docs in a separate PR and land that before this one (and only do the new cursor API), that might be best.

Reply:
Addressed in #12375